docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
+            } catch (err) {
+                error = "newDataSet: " + err;
+                throw error;
+            }
 
-Also within the `diff` folder, create a folder named `data-sets`. Within the `data-sets` folder create `DataSets.definition.ts` and `DataSets.handler.ts`.
+            const regex = /\.|\(/gi; // Replace . and ( with /
+            const regex2 = /\)/gi;   // Replace ) with .
 
-`DataSets.definition.ts` should contain:
-```typescript
-import { ICommandDefinition } from "@zowe/imperative";
-
-export const DataSetsDefinition: ICommandDefinition = {
-    name: "data-sets",
-    aliases: ["ds"],
-    summary: "data sets to diff",
-    description: "diff the first data set with the second",
-    type: "command",
-    handler: __dirname + "/DataSets.handler",
-    positionals: [
-        {
-            name: "oldDataSet",
-            description: "The old data set",
-            type: "string"
-        },
-        {
-            name: "newDataSet",
-            description: "The new data set",
-            type: "string"
+            // convert the old data set name to use as a path/file
+            let file = oldDataSet.replace(regex, "/");
+            file = file.replace(regex2, ".") + "dat";
+            // Load the downloaded contents of 'oldDataSet'
+            const oldContent = readFileSync(`${file}`).toString();
+
+            // convert the new data set name to use as a path/file
+            file = newDataSet.replace(regex, "/");
+            file = file.replace(regex2, ".") + "dat";
+            // Load the downloaded contents of 'oldDataSet'
+            const newContent = readFileSync(`${file}`).toString();
+
+            return diff.createTwoFilesPatch(oldDataSet, newDataSet, oldContent, newContent, "Old", "New");
         }
-    ],
-    profile: {
-        required: ["zosmf"]
     }
-};
-```
+    ```
 
-`DataSets.handler.ts` should contain the following:
-```typescript
-import { ICommandHandler, IHandlerParameters, TextUtils, Session } from "@zowe/imperative";
-import { DataSetDiff } from "../../../api/DataSetDiff";
-
-export default class DataSetsDiffHandler implements ICommandHandler {
-    public async process(params: IHandlerParameters): Promise<void> {
-
-        const profile = params.profiles.get("zosmf");
-        const session = new Session({
-            type: "basic",
-            hostname: profile.host,
-            port: profile.port,
-            user: profile.user,
-            password: profile.pass,
-            base64EncodedAuth: profile.auth,
-            rejectUnauthorized: profile.rejectUnauthorized,
-        });
-        const resp = await DataSetDiff.diff(session, params.arguments.oldDataSet, params.arguments.newDataSet);
-        params.response.console.log(resp);
-    }
-}
+3. In the `src` directory, add the following code to the end of the `index.ts` file to make the API available for other developers to import:
 
-```
+    ```typescript
+    export * from "./api/DataSetDiff";
+    ```
```

Comment:
Since the user likely deleted the contents in `src/api` before this step...
Should we also advise that they delete the current `export * from "./api/Files";` in this `index.ts` file?

I guess we can add a note in the `Setting up the new sample plug-in project` section about this.

Otherwise, I believe they will have compilation/transpilation errors

Reply:
edited the appropriate step.